### ISSUE TITLE
(Julia) Warning: Client 1 quit with exit code 1 and signal 0

### ISSUE THREAD

**Issue by zhenwu0728** [jmbuhr/otter.nvim#168]
I Just started to use Quarto in neovim. But I get "Warning: Client 1 quit with exit code 1 and signal 0" and then Auto-completion/LSP stops working in `qmd` file. However, LSP works fine in `jl` files.
So I guess this is the place I should file an issue.

The error message I got in lsp.log is 
```
ERROR: Unknown method completionItem/resolve.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] dispatch_msg(x::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, dispatcher::JSONRPC.MsgDispatcher, msg::
Dict{String, Any})
   @ JSONRPC ~/.julia/packages/JSONRPC/P0G1p/src/
typed.jl:81
 [3] run(server::LanguageServerInstance)
   @ LanguageServer ~/.julia/packages/LanguageServer/0vsx2/src/languageserverinstance.jl:382
 [4] top-level scope
   @ none:31
```

I also found a related issue [here](https://github.com/julia-vscode/LanguageServer.jl/issues/1182), but I don't know how to fix it.

Thank you in advance!

**Comment by jmbuhr** [comment]
It is a bit unusual for a language server to just crash after receiving a request it can't handle, but what we can do on otter's side is implement dynamic capabilities, such that requests are only sent to servers that can handle the request method.

**Comment by zhenwu0728** [comment]
I've no idea how to do that. Can you please tell me how to implement the dynamic capabilities for Julia in otter?
> It is a bit unusual for a language server to just crash after receiving a request it can't handle, but what we can do on otter's side is implement dynamic capabilities, such that requests are only sent to servers that can handle the request method.

**Comment by jmbuhr** [comment]
I started working on it here https://github.com/jmbuhr/otter.nvim/pull/169

**Comment by zhenwu0728** [comment]
Hi @jmbuhr, just checking in if I can help in some way. If the PR is ready, I can go ahead and test if it works in Julia.

**Comment by jmbuhr** [comment]
So... I merged the PR, meaning otter will no longer send a request if no server attached to the corresponding otter buffer has the capabilities for the method of the request. This is all we can do from this side.

In the process I did notice that the julia language server is not that much fun to work with in general, partly because of the reason you already linked; it really likes crashing at the slightest sign of danger (instead of reporting for example a "method not found" and continuing to run, as most other language servers do), but others have discussed this already to no avail. To be fair, while the LSP specifies such a response code for no-implemented methods, it doesn't specify that a language server shouldn't also treat an unknown request as a signal to crash.
